### ISSUE TITLE
Details about _NewEmptyTensorOp?

### ISSUE THREAD

**Issue by Jacobew** [facebookresearch/maskrcnn-benchmark#1097]
## ❓ Questions and Help
@chengyangfu @LeviViana  Hi, I'm curious about the details of  _NewEmptyTensorOp in https://github.com/facebookresearch/maskrcnn-benchmark/blob/614e42759652e66103c21808e82265b17cc97214/maskrcnn_benchmark/layers/misc.py#L18.
I find that if maskrcnn_benchmark.layers.Conv2d is replaced with nn.Conv2d, then CUDNN_STATUS_BAD_PARAM will arise when the input to mask head is in the shape of [0, 256, *, *], which means there are no masks detected.
Could you please provide some details how this class handle empty input ?  Thanks.

**Comment by chengyangfu** [comment]
Hi @Jacobew, 
    This part of the code was not implemented by me. So the following is just my thought. 
    In the forward of this operator,  `new_empty` is used. According to PyTorch document, `new_empty` returns the uninitialized tensor. I do not think this operator will modify the memory. It basically just change the shape variable of the tensor. Therefore, no cuda/cudnn operation is ran. 

      I think this is the light-weight operation and also guarantee how the empty tensor will be handled. It's a pretty neat design.

**Comment by askerlee** [comment]
@chengyangfu thanks for your explanation. I now understand it partly. But I'm still confused: does Conv2d() defined here only return a random tensor? What's the purpose then?
